src/Sarif/Core/WebRequest.cs

```diff
@@ -112,30 +112,37 @@ private static Uri SynthesizeAbsoluteUriFrom(Uri uri)
             return uri;
         }
 
+        // In this query pattern, note how backtracking is disabled with (?>...) while parsing the
+        // parameter names and values. A parameter name can't contain a '=', and a parameter value
+        // can't contain a '&', so it's never necessary to backtrack over an individual parameter
+        // name or value.
         const string QueryPattern =
             @"^
-              \?                              # A literal '?', followed by zero or more
-              (?<query>
+              \?                        # A literal '?', followed by zero or more of...
+              (
+                (?>
+                  (?<name>[^=]+)        # a parameter name (everything that's not an = sign),
+                )
+                =                       # an equal sign, and
                 (
-                  (?<name>[^=]+)                # parameter name (everything that's not an = sign),
-                  =                             # an equal sign, and
-                  (
-                    (?<value>[^&]*)             # the value (everything that's not an '&')...
-                    &?                          # and an '&' (except after the last one).
+                  (?>
+                    (?<value>[^&]*)     # the value (everything that's not an '&')...
                   )
-                )*
-                |                               # RFC 3986 does _not_ require a URI's query to
```

Comment:
>| [](start = 16, length = 1)

This alternation was not necessary, because if the query string didn't match the regex, we simply didn't populate the parameters dictionary. Removing it further improves performance (the engine never has to backtrack to the beginning to try the (trivial) alternative).

@mikefan Note that this is a PR comment, not a code comment, because the code it describes is gone.